src/setup-server.ts

```diff
@@ -15,6 +18,26 @@ export async function setupServer() {
 
   const { url } = await startStandaloneServer(server, {
     listen: { port: 4000 },
+    context: async ({ req }) => {
+      const token = req.headers.authorization;
+      try {
+        const decodedToken = jwt.verify(token, process.env.TOKEN_SECRET) as {
+          id: number;
+          iat: number;
+          exp: number;
+        };
```

Comment:
```suggestion
        const decodedToken = jwt.verify(token, process.env.TOKEN_SECRET) as JwtPayload;
```

Se vc consultar a tipagem da `@types/jsonwebtoken`, vai ver que existe essa interface JwtPayload que cobre o que vc quer tipar aqui:

<img width="989" alt="image" src="https://github.com/indigotech/onboard-tassyla-lima/assets/6992731/ee2384f2-5219-4833-b294-5bdb74d00de2">
*Segundo overload

<img width="737" alt="image" src="https://github.com/indigotech/onboard-tassyla-lima/assets/6992731/52004309-3713-4e88-b3c5-c8dd5a36e9c1">

Reply:
O "as" é pra vc afirmar que está retornando o JwtPayload em vez de uma string (poderia ser string tbm dependendo de como vc utilizasse o método).

Reply:
Vale atualizar nos testes tbm.

Reply:
Que legal, fica bem mais anxuto assim mesmo